examples/c/src/c/phi3.cpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <string>
+#include <cstring>
+#include "ort_genai.h"
+#include <thread>
+#include <csignal>
+#include <atomic>
+#include <functional>
+#include "../common.h"
+
+static TerminateSession catch_terminate;
+
+void signalHandlerWrapper(int signum) {
+  catch_terminate.signalHandler(signum);
+}
+
+// C API Example
+
+void CheckResult(OgaResult* result) {
```

Comment:
Why do we have these C++ using C API examples? They look like they're just inlining our C++ header vs using it.

Also, it looks like we're leaking objects, like the generator? I don't see OgaDestroyGenerator used (I didn't check the other cases).

Reply:
As discussed offline, I have kept only two C API examples with a disclaimer to encourage users to use C++ API.

Fixed the leaking memory issue, now I don't see any error while exiting.